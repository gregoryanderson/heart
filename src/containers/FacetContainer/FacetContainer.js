import React from "react";
import "./FacetContainer.css";
import PropTypes from "prop-types";
import FacetBtn from "../FacetBtn/FacetBtn";

export const FacetContainer = props => {
  const buttons = props.facets.facets.map(facet => {
    return (
      <FacetBtn
        assignment={facet.key}
        key={facet.value + facet.key}
        name={props.name}
        handleSearch={props.handleSearch}
      />
    );
  });
  return <section> {buttons} </section>;
};

export default FacetContainer;

FacetContainer.propTypes = {
  facets: PropTypes.object,
  name: PropTypes.string,
  handleSearch: PropTypes.func
};
