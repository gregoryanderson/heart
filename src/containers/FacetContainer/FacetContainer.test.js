import React from "react";
import { shallow } from "enzyme";
import { FacetContainer } from "./FacetContainer";

describe("FacetContainer", () => {
  let wrapper, mockHandleSearch, mockName, mockFacets;
  beforeEach(() => {
    mockHandleSearch = jest.fn();

    mockName = "color";

    mockFacets = {
      facets: [
        {
          0: {},
          1: {}
        }
      ]
    };

    wrapper = shallow(
      <FacetContainer
        handleSearch={mockHandleSearch}
        name={mockName}
        facets={mockFacets}
      />
    );
  });

  it("should  match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
