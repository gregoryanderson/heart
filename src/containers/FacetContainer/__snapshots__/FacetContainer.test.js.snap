// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacetContainer should  match the snapshot 1`] = `
<section>
   
  <FacetBtn
    handleSearch={[MockFunction]}
    key="NaN"
    name="color"
  />
   
</section>
`;
